File: Day - 4 (Class vs. Instance)/try.c
#include <stdio.h>

struct Person {
    int age;
};

void initPerson(struct Person *p, int initialAge) {
    // Add some more code to run some checks on initialAge
    if (initialAge > 0) {
        p->age = initialAge;
    } else {
        printf("Age is not valid, setting age to 0.\n");
        p->age = 0;
    }
}

void amIOld(struct Person *p) {
    // Write code determining if this person's age is old and print the correct statement:
    if (p->age < 13) printf("You are young.\n");
    else if (p->age < 18) printf("You are a teenager.\n");
    else printf("You are old.\n");
}

void yearPasses(struct Person *p) {
    // Increment this person's age.
    p->age++;
}

int main() {
    int T, age;
    scanf("%d", &T);

    for (int i = 0; i < T; i++) {
        scanf("%d", &age);
        struct Person p;
        initPerson(&p, age);
        amIOld(&p);

        for (int j = 0; j < 3; j++) {
            yearPasses(&p);
        }

        amIOld(&p);
        printf("\n");
    }

    return 0;
}

File: Day - 4 (Class vs. Instance)/try.py
class Person:
    def __init__(self, initial_age):
        # Add some more code to run some checks on initial_age
        if initial_age > 0:
            self.age = initial_age
        else:
            print("Age is not valid, setting age to 0.")
            self.age = 0

    def am_i_old(self):
        # Write code determining if this person's age is old and print the correct statement
        if self.age < 13:
            print("You are young.")
        elif self.age < 18:
            print("You are a teenager.")
        else:
            print("You are old.")

    def year_passes(self):
        # Increment this person's age
        self.age += 1

def main():
    T = int(input())
    for _ in range(T):
        age = int(input())
        p = Person(age)
        p.am_i_old()

        for _ in range(3):
            p.year_passes()

        p.am_i_old()
        print()

if __name__ == "__main__":
    main()
